addon/actions/action.js

```diff
@@ -88,13 +89,28 @@ export default emberObject.extend({
   },
 
   _promise() {
-    return this.get('adapter').ajax(this.get('url'), this.get('requestType'), this.get('data')).then((response) => {
-      if (this.get('config.pushToStore') && this._validResponse(response)) {
-        return this.get('serializer').pushPayload(this.get('store'), response);
-      } else {
-        return response;
-      }
-    });
+    return this.get('adapter')
+               .ajax(this.get('url'), this.get('requestType'), this.get('data'))
+               .then(this._onSuccess.bind(this), this._onError.bind(this));
+  },
+
+  _onSuccess(response) {
+    if (this.get('config.pushToStore') && this._validResponse(response)) {
+      return this.get('serializer').pushPayload(this.get('store'), response);
+    } else {
+      return response;
+    }
+  },
+
+  _onError(reason) {
+    if (this.get('config.pushToStore') && (reason instanceof InvalidError)) {
+      let model = this.get('model');
+      let store = this.get('store');
+      let errors = this.get('serializer').extractErrors(store, model.constructor, reason);
+      store.recordWasInvalid(model._internalModel, errors);
```

Comment:
I was thinking about the same but unfortunately it's PRIVATE :/ 
```
    @method recordWasInvalid
    @private
    @param {InternalModel} internalModel
    @param {Object} errors
  */
  recordWasInvalid(internalModel, errors) {
    internalModel.adapterDidInvalidate(errors);
  },
```

And as I was saying above, we can't assign errors to model in custom action because it will break an ember-data flow according to errors handling.

Let's imagine that we have a model Post and a custom action Publish. We are calling post.publish() and it will return an error eg "Publish action require passing publisher". Then we decided to update a title of the post. And unfortunately, after successful save our errors disappeared :/ Which should not be the case because we didn't publish successfully the post.

That's way they should stay separated from a model.